javascript/core/src/verbs/stepFunctions/simpleSteps.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { all, escape, op } from 'arquero'
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+
+import type { TableStep } from '../nodeFactories/index.js'
+import type {
+	AggregateArgs,
+	BinarizeArgs,
+	BooleanArgs,
+	DedupeArgs,
+	EraseArgs,
+	FillArgs,
+	FilterArgs,
+	FoldArgs,
+	GroupbyArgs,
+	ImputeArgs,
+	PivotArgs,
+	RecodeArgs,
+	RenameArgs,
+	RollupArgs,
+	SampleArgs,
+	SelectArgs,
+	SpreadArgs,
+	UnrollArgs,
+	WindowArgs,
+} from '../types/domainTypes.js'
+import { deriveBoolean } from '../util/expressions.js'
+import { compareAll, singleExpression } from '../util/index.js'
+import type { ExprFunctionMap } from '../util/types.js'
+
+export const aggregateStep: TableStep<AggregateArgs> = (
+	input,
+	{ groupby, column, operation, to },
+) => {
+	const expr = singleExpression(column, operation)
+	return input.groupby(groupby).rollup({ [to]: expr })
+}
+
+export const binarizeStep: TableStep<BinarizeArgs> = (
```

Comment:
How do we differentiate sync and async steps now with the elimination of promises? I.e., fetch returns a promise, but the rest of these would be async only within an observable context, right?

Reply:
Is the StepNode factory how promises are consistently chained?

Reply:
The execution steps can be either `async` or `sync`. We avoid `await` in the node implementations so that if data is available synchronously it will compute without an extra `tick()`. If you look for `handleMaybeAsync`, you'll see where that happens.

Reply:
The `stepNodeFactory` creates a Node Factory out of a step function.  The chaining has to be explicit with this model. I'd take a look at `graph.spec.ts` to see that in action